doc/source/reference/c-api/array.rst

```diff
@@ -761,29 +761,79 @@ cannot not be accessed directly.
 
 .. versionchanged:: 2.0
     Prior to NumPy 2.0 the ABI was different but unnecessary large for user
-    DTypes.  These accessors were all added in 2.0.
+    DTypes.  These accessors were all added in 2.0 and can be backported
+    (see :ref:`_migration_c_descr`).
+
+.. c:function:: npy_intp PyDataType_ELSIZE(PyArray_Descr *descr)
+
+    The element size of the datatype (``itemsize`` in Python).
+
+    .. note::
+        If the ``descr`` is attached to an array ``PyArray_ITEMSIZE(arr)``
+        can be used and is available on all NumPy versions.
+
+.. c:function:: void PyDataType_SET_ELSIZE(PyArray_Descr *descr, npy_intp size)
+
+    Allows setting of the itemsize, this is *only* relevant for string/bytes
+    datatypes as it is the current pattern to define one with a new size.
+
+.. c:function:: npy_intp PyDataType_ALIGNENT(PyArray_Descr *descr)
+
+    The alignment of the datatype.
 
 .. c:function:: PyObject *PyDataType_METADATA(PyArray_Descr *descr)
 
     The Metadata attached to a dtype, either ``NULL`` or a dictionary.
 
 .. c:function:: PyObject *PyDataType_NAMES(PyArray_Descr *descr)
 
-    ``NULL`` or a list of structured field names attached to a dtype,
-    this list should not be mutated, NumPy may change the way fields are
-    stored in the future.
+    ``NULL`` or a tuple of structured field names attached to a dtype.
 
 .. c:function:: PyObject *PyDataType_FIELDS(PyArray_Descr *descr)
 
     ``NULL``, ``None``, or a dict of structured dtype fields, this dict must
     not be mutated, NumPy may change the way fields are stored in the future.
 
+    This is the same dict as returned by `np.dtype.fields`.
+
 .. c:function:: NpyAuxData *PyDataType_C_METADATA(PyArray_Descr *descr)
 
     C-metadata object attached to a descriptor.  This accessor should not
     be needed usually.  The C-Metadata field does provide access to the
     datetime/timedelta time unit information.
 
+.. c:function:: PyArray_ArrayDescr *PyDataType_SUBARRAY(PyArray_Descr *descr)
+
+    Information about a subarray dtype eqivalent to the Python `np.dtype.base`
+    and `np.dtype.shape`.
+
+    If this is non- ``NULL``, then this data-type descriptor is a
+    C-style contiguous array of another data-type descriptor. In
+    other-words, each element that this descriptor describes is
+    actually an array of some other base descriptor. This is most
+    useful as the data-type descriptor for a field in another
+    data-type descriptor. The fields member should be ``NULL`` if this
+    is non- ``NULL`` (the fields member of the base descriptor can be
+    non- ``NULL`` however).
+
+    .. c:type:: PyArray_ArrayDescr
```

Comment:
Is this indented on purpose? See [the rendered page](https://output.circle-artifacts.com/output/job/eff94025-1729-4f27-aca0-11e307460ad7/artifacts/0/doc/build/html/reference/c-api/array.html#c.PyDataType_SUBARRAY)